File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int first = m - 1, second = n - 1, index = m + n - 1;
        while (second >= 0) {
            if (first >= 0 && nums1[first] > nums2[second])
                nums1[index--] = nums1[first--];
            else
                nums1[index--] = nums2[second--];
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    p1, p2, p3 := m - 1, n - 1, m + n - 1

    for p2 >= 0 {
        if p1 >= 0 && nums1[p1] > nums2[p2] {
            nums1[p3] = nums1[p1]
            p1--
        } else {
            nums1[p3] = nums2[p2]
            p2--
        }
        p3--
    }
}